components/brave_wallet/browser/zcash/zcash_orchard_storage.cc

```diff
@@ -418,4 +580,869 @@ std::optional<ZCashOrchardStorage::Error> ZCashOrchardStorage::UpdateNotes(
   return std::nullopt;
 }
 
+base::expected<std::optional<uint32_t>, ZCashOrchardStorage::Error>
+ZCashOrchardStorage::GetLatestShardIndex(mojom::AccountIdPtr account_id) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+
+  if (!EnsureDbInit()) {
+    return base::unexpected(
+        Error{ErrorCode::kDbInitError, database_.GetErrorMessage()});
+  }
+
+  sql::Statement resolve_max_shard_id(
+      database_.GetCachedStatement(SQL_FROM_HERE,
+                                   "SELECT "
+                                   "MAX(shard_index) FROM " kShardTree " "
```

Comment:
Having MAX() most likely requires indexes. Please add a TODO or an issue to research that